coalib/bears/Bear.py

```diff
@@ -45,6 +46,26 @@ def _is_debugged(bear):
             map(str.lower, bear.section['debug_bears']))
 
 
+def _is_profiled(bear):
+    """
+    Check whether the bear is in profile mode according to its section-settings.
+
+    :param bear: Bear object.
+    :return:     True if ``profile`` is ``True`` or False if ``profile`` is
+                 ``False`` else return directory path specified in ``profile``.
+    """
+    if not isinstance(bear, Bear):
+        raise ValueError('Positional argument bear is not an instance of '
+                         'Bear class.')
+    if 'profile' not in bear.section:
+        return False
+    try:
+        return bool(bear.section['profile'])
+    except ValueError:
+        pass
+    return bear.section['profile'].value
```

Comment:
If you squint carefully, you will see this is identical to `_is_debugged` except for this last line.

If your helper function returns the `Setting`, i.e. `bear.section['x']` at the end, you can move the custom logic of looking inside the setting to the section of code which needs to understand the setting value.  the caller would then look for True, False, or Setting.

Reply:
I initially thought of that but intentionally I dropped it because not only the last line which are different, but if you see this logic
```python
    if 'debug_bears' not in bear.section:
        return False
```
and 
```python
    if 'profile' not in bear.section:
        return False
```
If I try to make a helper function which work for both then it make both the condition dependent as if any of the conditon will be true both will run, so yeah its little bit complicated and it will become somewhat more complex.

Reply:
I don't see a problem unifying the code ^^

> If I try to make a helper function which work for both then it make both the condition dependent as if any of the conditon will be true both will run

Why should that happen? You still have an `_is_profiled` function, but it will call some code both common for `_is_debugged` and `_is_profiled`^^

Reply:
I do see this logic
```py
    if 'debug_bears' not in bear.section:
        return False
```
and
```py
    if 'profile' not in bear.section:
        return False
```
.. and I see they are absolutely identical.  If you are seeing a difference, you are not squinting correctly.

To make sure there is no confusion, the new function would have a signature like `_setting_is_enabled(setting_name)` .  that helper would only look at one setting name; the name provided in the args